src/core/execute-serialize.c

```diff
@@ -1631,18 +1631,18 @@ static int exec_context_serialize(const ExecContext *c, FILE *f) {
         if (c->root_image_options) {
                 _cleanup_free_ char *options = NULL;
 
-                LIST_FOREACH(mount_options, o, c->root_image_options) {
-                        if (isempty(o->options))
+                for (PartitionDesignator i = 0; i < _PARTITION_DESIGNATOR_MAX; i++) {
+                        if (isempty(c->root_image_options->options[i]))
```

Comment:
hmm, here you use isempty(), but elsewhere you just check for non-NULL-ness... might make sense to go for isempty() everyhwere.